neuralpp/test/quick_tests/symbolic/expression_test.py

```diff
@@ -140,25 +141,27 @@ def test_basic_function_application():
 def test_basic_quantifier_expressions():
     from neuralpp.symbolic.constants import int_add, int_multiply
     i = BasicVariable('i', int)
-    i_range = (0 < i) & (i < 10)
+    context = Z3SolverExpression()
+    i_range = context & (0 < i) & (i < 10)
     sum_ = BasicSummation(int, i, i_range, i)
     assert sum_.subexpressions[0].syntactic_eq(int_add)
     assert sum_.subexpressions[1].syntactic_eq(i)
-    assert sum_.subexpressions[2].syntactic_eq(i_range)
+    assert sum_.subexpressions[2].structure_eq(i_range)  # cannot check syntactic_eq() of Z3SolverExpression.
```

Comment:
changed a bunch of `syntactic_eq` to `structure_eq` because `syntactic_eq()` does not work for `Z3SolverExpression` (for Z3's `syntactic_eq()`, we compare the `z3_object`; `Z3SolverExpression` does not contain a `z3_object` to begin with, so all its `syntactic_eq` calls return `False`)